src/core/processing/qgsprocessingcontext.cpp

```diff
@@ -85,6 +85,17 @@ void QgsProcessingContext::setInvalidGeometryCheck( QgsFeatureRequest::InvalidGe
       break;
     }
 
+    case QgsFeatureRequest::GeometryFixInvalid:
+    {
+      auto callback = [ = ]( const QgsFeature & feature )
+      {
+        if ( mFeedback )
+          mFeedback->reportError( QObject::tr( "Feature (%1) has invalid geometry and has been fixed." ).arg( feature.id() ) );
```

Comment:
Should be information, not error

Reply:
I was thinking to make it a `pushInfo` call after seeing that there's no `pushWarning`. Error seems to be a bit of an overkill.

Reply:
I think error is fine as it displays the text in red in the processing log. We display messages about the lack of index that may cause the process to be slower, which is less dramatic than changing in place a geometry IMO.

Reply:
Ok, let's leave it as is and instead add a pushWarning todo to our mental todo stack :+1: